mathesar_ui/src/systems/table-view/table-inspector/record-summary/AppendColumn.svelte

```diff
@@ -1,13 +1,29 @@
 <script lang="ts">
   import { ButtonMenuItem, DropdownMenu } from '@mathesar-component-library';
+  import { getTabularDataStoreFromContext } from '@mathesar/stores/table-data';
   import type { Column } from '@mathesar/api/types/tables/columns';
-  import { iconAddNew } from '@mathesar/icons';
+  import { iconAddNew, iconTableLink } from '@mathesar/icons';
   import { getColumnIconProps } from '@mathesar/utils/columnUtils';
 
+  const tabularData = getTabularDataStoreFromContext();
+  $: ({ processedColumns } = $tabularData);
+
   export let columns: Column[];
   export let onSelect: (column: Column) => void;
 
+  function checkForForeignKey(column: Column) {
+    const linkFkType = $processedColumns.get(column.id)?.linkFk?.type;
+    if (linkFkType) {
+      return true;
+    }
+    return false;
+  }
+
   function getIcon(column: Column) {
+    const isForeignKey = checkForForeignKey(column);
+    if (isForeignKey) {
+      return iconTableLink;
+    }
```

Comment:
@seancolsen Looking at the implementation of `getColumnIconProps`, if there is a constraint type `primary` on the column then its icon takes precedence over the foreign key icon.

```
export function getColumnIconProps(
  _column: DisplayColumn,
): IconProps | IconProps[] {
  if (_column.constraintsType?.includes('primary')) {
    return iconConstraint;
  }

  if (_column.constraintsType?.includes('foreignkey')) {
    return iconTableLink;
  }

  ...
)
``` 
This behaviour is different from other dropdowns where foreign key icon takes precedence (as per my first [PR](https://github.com/centerofci/mathesar/pull/2750)).

**Do we want to give precedence to the primary icon?**

In that case we can modify `getColumnConstraintTypeByColumnId` in `columnUtils.ts` like this:
```
export function getColumnConstraintTypeByColumnId(columnId: number, processedColumns: Map<number, ProcessedColumn>) {
  const processedColumn = processedColumns.get(columnId);
  const constraintsType = Array.from(
    new Set(
      [
        ...(processedColumn?.exclusiveConstraints ?? []),
        ...(processedColumn?.sharedConstraints ?? []),
      ]
        .filter((constraintType) => !!constraintType)
        .map((constraintType) => constraintType.type),
    ),
  );
  return constraintsType;
}
```

Reply:
@hemant-hc yes, I think it makes sense to give precedence to the primary icon. Feel free to make those changes in this PR